apps/web/src/components/search/index.js

```diff
@@ -20,27 +20,28 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 import * as Icon from "../icons";
 import "./search.css";
 import Field from "../field";
+import { inlineDebounce } from "../../utils/debounce";
 
-function SearchBox(props) {
+function SearchBox({ onSearch }) {
   return (
     <Field
       data-test-id="search-input"
       autoFocus
       id="search"
       name="search"
       type="text"
-      sx={{m: 0,mx: 1,mt: 1}}
+      sx={{ m: 0, mx: 1, mt: 1 }}
       placeholder="Type your query here"
-      onKeyDown={(e) => {
-        if (e.key === "Enter") props.onSearch(e.target.value);
-      }}
+      onChange={(e) =>
+        inlineDebounce("search", () => onSearch(e.target.value), 250)
+      }
```

Comment:
1. Why are we using `inlineDebounce` here instead of just `debounce`?
2. Doesn't onChange get called on focus away? Not sure.

Reply:
1. Do it like this: `onChange={debounce(...)}`
2. Ah, okay.

Reply:
Ok